x-pack/plugins/cases/public/components/markdown_editor/eui_form.tsx

```diff
@@ -60,14 +72,49 @@ export const MarkdownEditorForm = React.memo(
         [id, field.value, caseTitle, caseTags]
       );
 
+      useEffect(() => {
+        const storageDraftComment = draftCommentStorageKey && storage.get(draftCommentStorageKey);
+        if (storageDraftComment && storageDraftComment !== '') {
+          field.setValue(storageDraftComment);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+      }, []);
+
+      useEffect(() => {
+        if (initialValue && initialValue !== field.value) {
+          setShowVersionConflictWarning(true);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+      }, [initialValue]);
+
+      useDebounce(
+        () => {
+          if (isFirstRender.current) {
+            isFirstRender.current = false;
+            return;
+          }
+          if (draftCommentStorageKey) {
+            if (field.value !== '') {
+              storage.set(draftCommentStorageKey, field.value);
+            } else {
+              storage.remove(draftCommentStorageKey);
+            }
+          }
+        },
+        STORAGE_DEBOUNCE_TIME,
+        [field.value]
+      );
```

Comment:
I think it is best if we do the following:

- Extract the logic in a hook called `useMarkdownSessionStorage` and use it here. 
- Use [`useSessionStorage`](https://github.com/streamich/react-use/blob/master/docs/useSessionStorage.md) from the [`react-use`](https://github.com/streamich/react-use) library.
- Remove unnecessary `useEffect`
- Use `useRef` to save the initial value.

Example

```
const useMarkdownSessionStorage = ({ field, sessionKey, initialValue }) => {
  const [hasConflicts, setHasConflicts] = useState<boolean>(false);
  const isFirstRender = useRef(true);
  const initialValueRef = useRef(initialValue);

  const [sessionValue, setSessionValue] = useSessionStorage<string>(
    sessionKey ?? 'cases.markdown' // maybe we can make it required and store the description when you add a case
  );

  if (!isEmpty(sessionValue) && isFirstRender.current) {
    isFirstRender.current = false;
    field.setValue(sessionValue);
  }

  if (initialValue !== initialValueRef.current && initialValue !== field.value) {
    initialValueRef.current = initialValue;
    setHasConflicts(true);
  }

  useDebounce(
    () => {
      if (isFirstRender.current) {
        isFirstRender.current = false;
        return;
      }

      if (sessionKey) {
        if (field.value !== '') {
          setSessionValue(field.value);
        } else {
          setSessionValue('');
        }
      }
    },
    STORAGE_DEBOUNCE_TIME,
    [field.value]
  );

  return { hasConflicts };
};

```

Reply:
Update: Not able to reproduce the issue again